### ISSUE TITLE
break from inside a function, not inside a loop does not behave as indicated in the manual 

### ISSUE THREAD

**Issue by james-d-mitchell** [gap-system/gap#298]
In the function:

    foo := function(x)
      if x = 0 then 
        Print("this is an error");
        break;
      else
         y := 1;
      fi;
      return y;
   end;

I would expect that several things are wrong. First, `break` is not in a loop and so an error should be given. Second, in the case that `x` is `0` we should get another error saying that `y` is not bound. 

This is not what happens, GAP seems to leave the function at the break statement. 

    gap> foo(0);
    this is an error

This is either a bug in GAP or in the documentation of `break`.

**Comment by ChrisJefferson** [comment]
This also happens with `continue`.

It's not too hard to fix this -- we can add a check in the DoExecFunc family of functions to check if we are leaving the function because of a `break` or `continue`, and if so produce an `Error`. Then the only question is -- do we want to fix GAP's behaviour, or the documentation?

**Comment by fingolfin** [comment]
To me this is a bug in GAP's behavior, which should be fixed; the documentation should NOT be modified to accommodate for this.

**Comment by alex-konovalov** [comment]
I agree with @fingolfin. BTW, in the GAP prompt is works as documented:

```
gap> break;
Error, A break statement can only appear inside a loop
not in any function at line 3 of *stdin*
gap> continue;
Error, A continue statement can only appear inside a loop
not in any function at line 4 of *stdin*
```

**Comment by stevelinton** [comment]
Yes. It should be an error to call break if we are not in a loop within the currently executing function.
Arguably it's a syntax error even to code the break in that context.

I will look at the interpreter.

**Comment by stevelinton** [comment]
#304 should fix this. It makes it a syntax error to have a break or continue not in a loop, so some code which had such a statement but never executed it might now fail to parse, but I think this is the best solution.

In particular it adds the work of tracking loop nesting depth when code is being read, rather than when it is being executed, which seems preferable.

**Comment by fingolfin** [comment]
PR #324 was merged, instead of #304; so I assume this can be closed now?
